### PR TITLE
Introducing -e to bail out early, checking variables. (smoe:quick_feedback)

### PR DESCRIPTION
**Pull request by smoe**
Hello,
just to give me  (or you when you get an error report) some extra confidence not to run into earlier errors, I have introduced the "set -e" at all (most) the bash scripts that also have a shebang line. 
It is fairly easy to start your tool with a boggled config file or a missing userdata file. This then runs with sudo but undefined variables, which is dangerous, so I introduced a few tests.
Many thanks!

### CODE REVIEW

**Comment by pyavitz** [comment]
I have no issue accepting this. But I would like it to be tested on either Jammy or Bookworm before merging. Trixie isn't currently supported and `set -e` in the past has produced false positives in the builder. Of course; the past was a million years ago and the code is a lot cleaner now. I would imagine it's probs fine.

**Comment by smoe** [comment]
Very reasonable - no urge, obviously. For the longer term please also consider to have the code pass the checks by https://www.shellcheck.net/.

File: lib/function/miscellaneous [hunk 1]
```diff
@@ -217,8 +211,7 @@ if [[ -f "/root/userscripts/uscripts" ]]; then
 		echo -e "\033[1mDone\e[0m."
 	fi
 fi
-if [[ -e "/root/userscripts" ]]; then rm -fdr /root/userscripts; fi
-}
+rm -fdr /root/userscripts; fi
```
Comment (pyavitz) [inline lib/function/miscellaneous]: This appears to break a function and you left an `; fi` which would no longer be required.

**Comment by pyavitz** [comment]
Alright. After checking further, adding `set -e` to every function file seems counter productive, plus `set -eu -o pipefail` is already included and in the scripts that run those functions, except the kernel builds. As that's where `set -e` can produce false fails and kill builds over warnings.

The changes in the menu bits are fine. Although if people did as the README suggested there wouldn't be a need for the builder to babysit dialog. But that's whatever I suppose.

**Comment by smoe** [comment]
Fixed the syntax - sorry.
Instead of 
```
#!/bin/bash
set -e
```
both our concerns may be addressed by rewriting as
````
#!/bin/bash -e
```
so the set -e would only be executed when the shebang line is not interpreted as a mere comment?

**Comment by pyavitz** [comment]
> #!/bin/bash -e

To what end exactly? What is it that you think the shebang inside the function file is doing that is creating an issue? Besides the fact it makes the code POP when reviewing and editing in an editor, I'm not seeing how them being there are causing the scripts to act out. Am I missing something?

**Comment by smoe** [comment]
The shebangs are perfectly fine. I just observed follow-up errors on my side when the configuration had failed because of a missing dependency, which a "set -e" in place would have prevented.  That is why I am after those "set -e"s so much, but you said you addressed that if starting your scripts correctly, so I happily accept that.

For all the source commands the shebang line is not needed (which had led me to trying a direct execution).   But when the direct execution is performed, and thus the shebang line is interpreted, then the "set -e" should be active to protect your users (to some degree). So I thought the set -e (and happily also the -u and -o pipefail) could move there.

File: lib/function/miscellaneous (hunk 1 above)
Comment (smoe) [reply lib/function/miscellaneous]: Fixed.

**Comment by pyavitz** [comment]
> The shebangs are perfectly fine. I just observed follow-up errors on my side when the configuration had failed because of a missing dependency, which a "set -e" in place would have prevented. That is why I am after those "set -e"s so much, but you said you addressed that if starting your scripts correctly, so I happily accept that.
> 

Well it has been addressed to a certain degree. I don't believe I have them in any `lib/dialog/` of the dialog scripts. I can see no reason why `set -e` can't be added there.

> For all the source commands the shebang line is not needed (which had led me to trying a direct execution). But when the direct execution is performed, and thus the shebang line is interpreted, then the "set -e" should be active to protect your users (to some degree). So I thought the set -e (and happily also the -u and -o pipefail) could move there.

`set -eu -o pipefail` is in both staging and rootfs scripts, so adding `set -e` to the function files which are already being `source` would be redundancy.

The shebang inside the function and board files are just there to make the text POP. They serve no purpose other than that.

**Pull request closed without merging**